### DISCUSSION TITLE
RealTensorValue in RotationMatrix

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#16140] [Q&A General] [answered]
For a `RealTensorValue` type, do you fill in values moving left to right across rows (how arrays are usually done), or do you fill down each column? 

I ask because I'm looking at the rotation matrix from euler angles computed [here](https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/src/utils/RotationTensor.C#L43). It looks like it is following the [Z1X2Z3 convention](https://en.wikipedia.org/wiki/Euler_angles#Rotation_matrix) as expected, but for instance the 2nd entry (row 1 column 2) isn't going in the 2nd entry for the variable, it's going in the 4th (index 3), which is where would expect column 1 row 2, unless it really does need to be filled in top to bottom. Or perhaps the function is just doing the transpose?

**Comment by aeslaughter** [comment]
@dschwen Do you know the answer to this?

**Reply by dschwen** [reply]
@SudiptaBiswas worked with Euler angle rotations in her sintering work.

**Reply by jessecarterMOOSE** [reply]
@SudiptaBiswas Just curious if you know the answer to this. It looks like the rotation matrix being built in https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/src/utils/RotationTensor.C is the _transpose_ of what you would expect based on the vector indices being stored, assuming you enter in the values row-wise (index 0 is R11, index 1 is R12, etc). Depending on the application, the transpose may be what you want, but I just want to be clear what's going on. The Euler angles for a rotation matrix and its transpose will be different.

**Comment by jessecarterMOOSE** [comment, marked as answer]
In case anybody finds this, I think the answer is clear and I just wasn't seeing it. As the documentation clearly stated, the Bunge convention is used for the Euler angles, meaning the angles are used to build the _passive_ (sample-to-crystal) rotation matrix, so yes, the transpose of the matrix is computed and applied when using Euler angles.

The [documentation](https://mooseframework.inl.gov/source/materials/ComputeElasticityTensor.html) has been updated a bit. See also https://github.com/idaholab/moose/pull/16341. I added some functionality such that one can apply a rotation matrix directly to the elasticity tensor.
